File: ifeeds_app/lib/models/feedback_model.dart
import 'dart:convert';

import 'package:ifeeds_app/models/feedback_avaliacao_model.dart';
import 'package:intl/intl.dart';

class FeedbackModel {
  int? id;
  String? codigo;
  String? titulo;
  String? descricao;
  String? matricula;
  double? nota;
  FeedbackAvaliacaoModel? avaliacao;
  String? campus;
  String? resposta;
  bool? status;
  DateTime? dataCriacao;
  DateTime? dataResposta;

  FeedbackModel({
    this.id,
    this.codigo,
    this.titulo,
    this.descricao,
    this.matricula,
    this.nota,
    this.avaliacao,
    this.campus,
    this.resposta,
    this.status,
    this.dataCriacao,
    this.dataResposta,
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'codigo': codigo,
      'titulo': titulo,
      'descricao': descricao,
      'matricula': matricula,
      'nota': nota,
      'avaliacao': avaliacao?.toMap(),
      'campus': campus,
      'resposta': resposta,
      'status': status,
      'dataCriacao': new DateFormat("yy-MM-dd").format(dataCriacao!),
      'dataResposta': dataResposta == null
          ? null
          : new DateFormat("yy-MM-dd").format(dataCriacao!),
    };
  }

  factory FeedbackModel.fromMap(Map<String, dynamic> map) {
    return FeedbackModel(
      id: map['id'],
      codigo: map['codigo'],
      titulo: map['titulo'],
      descricao: map['descricao'],
      matricula: map['matricula'],
      nota: map['nota'],
      avaliacao: FeedbackAvaliacaoModel.fromMap(map['avaliacao']),
      campus: map['campus'],
      resposta: map['resposta'],
      status: map['status'],
      dataCriacao: DateFormat('yy-MM-dd').parse(map['dataCriacao']),
      dataResposta: map['dataResposta'] == null
          ? null
          : DateFormat('yy-MM-dd').parse(map['dataResposta']),
    );
  }

  factory FeedbackModel.fromJson(String source) =>
      FeedbackModel.fromMap(json.decode(source));
}

File: IfeedsApi/Api/Models/FeedbackModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;
using AutoMapper;
using IfeedsApi.Domain.Models;

namespace IfeedsApi.Api.Models
{
    [AutoMap(typeof(Feedback), ReverseMap = true)]
    [AutoMap(typeof(Usuario), ReverseMap = true)]
    public class FeedbackModel
    {
        public int Id { get; set; }
        public string Codigo { get; set; }

        public string Titulo { get; set; }

        public string Descricao { get; set; }

        public string Matricula { get; set; }

        public decimal Nota { get; set; }

        public FeedbackAvaliacaoModel Avaliacao { get; set; }


        public string Campus { get; set; }

        public string Resposta { get; set; }

        public bool Status { get; set; }

        public DateTime DataCriacao { get; set; }

        public DateTime? DataResposta { get; set; }

    }

    [AutoMap(typeof(Feedback), ReverseMap = true)]
    public class FeedbaackModelRequest
    {
        [Required(ErrorMessage = "Título é um campo obrigatório")]
        [MaxLength(45)]
        public String Titulo { get; set; }


        [Required(ErrorMessage = "Descrição é um campo obrigatório")]
        public String Descricao { get; set; }


        [Required(ErrorMessage = "Nota é um campo obrigatório")]
        [Range(0, 5)]
        public decimal Nota { get; set; }


        public int AvaliacaoId { get; set; }

        public int UsuarioId { get; set; }

    }
}
